.github/workflows/release-source.yml

```
@@ -25,283 +27,186 @@ jobs:
+            exit 1
+          elif [ "$input_major" -lt "$file_major" ]; then
+            echo "Error: Major version cannot be less than the current version."
+            exit 1
+          elif [ "$input_minor" -lt "$file_minor" ]; then
+            if [ "$input_major" -eq "$max_major" ] && [ "$input_minor" -eq 0 ] && [ "$input_patch" -eq 0 ]; then
+              echo "Valid version input"
+            else
+              echo "Error: Minor version cannot be less than the current version."
+              exit 1
             fi
-            echo "test"
-          done
-          readme="$readme$commit_titles\n\n### Release dependencies first\n\nIn case of dependent visyn repositories follow dependency tree from the top.\n\n"
-          readme="$readme### 🏁 Finish line\n\n"
-          readme="$readme* [ ] Inform colleagues and customers about the release\n"
-          readme="$readme* [ ] Celebrate the new release 🥳"
-          echo -e "$readme"
-          echo "commits=$commit_titles"
-          echo "readme=$readme" >> "$GITHUB_OUTPUT"
-          echo "commits=$commit_titles" >> "$GITHUB_OUTPUT"
-      - name: setup git config
+          elif [ "$input_patch" -lt "$file_patch" ]; then
+            if [ "$input_major" -eq "$max_major" ] && [ "$input_minor" -eq 0 ] && [ "$input_patch" -eq 0 ]; then
+              echo "Valid version input"
+            elif [ "$input_major" -eq "$file_major" ] && [ "$input_minor" -eq "$max_minor" ] && [ "$input_patch" -eq 0 ]; then
+              echo "Valid version input"
+            else
+              echo "Error: Patch version cannot be less than the current version."
+              exit 1
+            fi
+          fi
+          echo "New version is $input_major.$input_minor.$input_patch"
+          echo "RELEASE_VERSION=$input_release_version" >> "$GITHUB_ENV"
+          
+      - name: create release branch
         run: |
           git config user.name "$GITHUB_ACTOR"
           git config user.email "<>"
-        env:
-          GITHUB_ACTOR: ${{ github.actor }}
-      - name: push branch
-        run: |
-          echo "push branch"
-          git commit -am "[skip ci]Prepare release v$RELEASE_VERSION"
-          git push origin "$BRANCH_NAME"
-        env:
-          RELEASE_VERSION: ${{ steps.get-release-version.outputs.release_version }}
-          BRANCH_NAME: ${{ steps.get-release-version.outputs.new_branch_name }}
-      - name: trigger build
-        uses: datavisyn/github-action-trigger-workflow@v1
-        with:
-          owner: ${{ github.repository_owner }}
-          repo: ${{ github.event.repository.name }}
-          github_token: ${{ github.token }}
-          workflow_file_name: "build.yml"
-          ref: "${{ steps.get-release-version.outputs.new_branch_name }}"
-          github_user: ${{ env.GITHUB_USER }}
-      - name: create pull request
+          git checkout -b release/"$RELEASE_VERSION"
```

```suggestion
          git checkout -b release-"$RELEASE_VERSION"
```
Please avoid `/` in branch names - they sometimes cause issues in GA

---

changes added